node/overseer/overseer-gen/proc-macro/src/impl_builder.rs

```diff
@@ -107,46 +61,379 @@ pub(crate) fn impl_builder(info: &OverseerInfo) -> proc_macro2::TokenStream {
 		})
 		.collect::<Vec<_>>();
 
-	let generics = quote! {
-		< S, #( #baggage_generic_ty, )* >
-	};
-	let where_clause = quote! {
+	// A helper structure to carry fields of a builder to produce setters
+	struct BuilderFieldHelper {
```

Comment:
This tries to unify across baggage fields (which might or might not have relevant generic) and subsystems (which have generic) which I don't think is a good idea. After all they have different trait bounds etc which causes the whole issue you mitigated by using `TokenStream`s.